gst/edge/edge_src.c

```diff
@@ -345,20 +378,21 @@ gst_edgesrc_create (GstBaseSrc * basesrc, guint64 offset, guint size,
     GstBuffer ** out_buf)
 {
   GstEdgeSrc *self = GST_EDGESRC (basesrc);
-
-  nns_edge_data_h data_h;
+  nns_edge_data_h data_h = NULL;
   GstBuffer *buffer = NULL;
   guint i, num_data;
   int ret;
 
   UNUSED (offset);
   UNUSED (size);
 
-  data_h = g_async_queue_pop (self->msg_queue);
+  while (self->playing && !data_h) {
+    data_h = g_async_queue_timeout_pop (self->msg_queue, G_USEC_PER_SEC);
```

Comment:
Q: timed-out param
Should we set the timeout to wait for new data? If edge device does not send new data in connected state, then this may raise an error.

Reply:
This is to prevent infinite waiting of data.
Even if timeout occurs while connected, it waits for an additional 1 second and no error occurs.